feat(wallet): lock/unlock all accounts at the same time

VmMad opened the pull request:
# Description of change

Unlock all accounts when unlocking a single one, refactor accounts in the wallet to avoid emitting refetch events and only send a single one when unlocking all of them. Update methods tha tunlock the source to unlock all accounts at the same time

## Links to any relevant issues

Fixes #5451

## How the change has been tested

- [ ] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

VmMad committed:
wip(wallet): make all accounts be unlocked when unlocking any

VmMad committed:
wip: more stuff

brancoder committed:
update mnemonic account

VmMad committed:
Co-authored-by: Bran <brancoder@users.noreply.github.com>

brancoder committed:
revert stored account

VmMad committed:
wip: trying to lock only when unlock

VmMad committed:
feat(wallet): improve event emission

VmMad committed:
refactor: cleanup all debug console logs

VmMad committed:
feat(wallet): when unlocking some account, also unlock all of them

VmMad committed:
Merge branch 'develop' into tooling-wallet/refactor-passwords

VmMad committed:
chore: cleanup and revert unnecessary changes

VmMad committed:
Merge branch 'develop' into tooling-wallet/refactor-passwords

VmMad committed:
fix: unlock accounts after creation

KeitoTadashi commented on apps/wallet/src/ui/app/components/accounts/UnlockAccountContext.tsx:
```diff
@@ -67,7 +53,7 @@ export function UnlockAccountProvider({ children }: UnlockAccountProviderProps)
             try {
-                await backgroundClient.lockAccountSourceOrAccount({ id: account.id });
+                await backgroundClient.lockAllAccounts({ id: account.id });
                 toast('Account locked');
```
```suggestion
                toast('Accounts locked');
```

VmMad committed:
fix: message

marc2332 commented on apps/wallet/src/ui/app/components/accounts/UnlockAccountContext.tsx:
```diff
@@ -67,8 +53,8 @@ export function UnlockAccountProvider({ children }: UnlockAccountProviderProps)
             try {
-                await backgroundClient.lockAccountSourceOrAccount({ id: account.id });
-                toast('Account locked');
+                await backgroundClient.lockAllAccounts({ id: account.id });
```
Not necessary I think?
```suggestion
                await backgroundClient.lockAllAccounts({ });
```

marc2332 replied:
just something I noticed while having a look

marc2332 commented:
@copilot can you explain to me how "skipEventEmit" is related to locking/unlocking all the accounts? On the surface its hard to see the connection

brancoder committed:
Merge branch 'develop' into tooling-wallet/refactor-passwords

brancoder committed:
feat(wallet): improve locking accounts (#9784)

# Description of change

improve account locking/unlocking

---------

Co-authored-by: JCNoguera <88061365+VmMad@users.noreply.github.com>

evavirseda commented on apps/wallet/src/ui/app/pages/accounts/manage/AccountGroupItem.tsx:
```diff
@@ -61,9 +61,11 @@ export function AccountGroupItem({
         e.stopPropagation();
         if (account.isLocked) {
-            unlockAccount(account);
+            if (account.isPasswordUnlockable) {
```
everything looks great and works smoothly 😊  j
ust one question: I'm not sure I see why we need this check 🙏

brancoder committed:
feat(wallet): improve accounts events (#9808)

# Description of change

- consolidates events for invalidating queries
- removes unused code
- decouples classes from emitting events responsible for UI changes

brancoder committed:
remove password check

brancoder committed:
remove password check

evavirseda reviewed (approved):
nice job 👯‍♂️

marc2332 reviewed (approved):
lgtm.
